### PR TITLE
ENG-2376: properly handling poll response of the new API

### PR DESCRIPTION
**Pull request by gReis89**
Screen Shot 2021-05-20 at 15 42 04

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **greis89**

If they are not already assigned, you can assign the PR to them by writing `/assign @greis89` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/state/component-repository/components/actions.js [hunk 1]
```diff
@@ -212,9 +212,16 @@ export const pollECRComponentInstallStatus = (componentCode, stepFunction) => di
...
-              { id: 'componentRepository.components.installRollback' },
+              res.payload.installErrorMessage,
+              TOAST_WARNING,
+            ));
+          }
+
+          if (res.payload.rollbackErrorMessage) {
+            dispatch(addToast(
+              res.payload.rollbackErrorMessage,
               TOAST_WARNING,
```
Comment (ichalagashvili) [inline src/state/component-repository/components/actions.js:225]: does toast show correctly the message here?

File: src/api/component-repository/components.js [hunk 2]
```diff
@@ -108,6 +109,7 @@ export const getECRComponentInstallPlan = code => (
     domain: '/digital-exchange',
     method: METHODS.GET,
     mockResponse: COMPONENT_INSTALL_PLAN,
+    // mockResponse: FAILED_INSTALL_PLAN,
```
Comment (ichalagashvili) [inline src/api/component-repository/components.js:112]: should we delete this?

File: src/api/component-repository/components.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ import {
   COMPONENT_UNINSTALLATION_COMPLETED,
   COMPONENT_USAGE_LIST,
   COMPONENT_INSTALL_PLAN,
+  // FAILED_INSTALL_PLAN,
```
Comment (ichalagashvili) [inline src/api/component-repository/components.js:10]: should we delete this?

**Review by ichalagashvili — COMMENTED** [review]
some questions only

File: src/state/component-repository/components/actions.js (hunk 1 above)
Comment (gReis89) [reply src/state/component-repository/components/actions.js:225]: it does! you can see the screenshot attached in this PR. The error message will come directly from the BE now.

File: src/api/component-repository/components.js (hunk 2 above)
Comment (gReis89) [reply src/api/component-repository/components.js:112]: yeah.. I thought about keeping it to know we have a FAILED mock response too, but I agree with you now, am going to remove it.

**Review by ichalagashvili — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/1075#pullrequestreview-665657509" title="Approved">ichalagashvili</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [ichalagashvili]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**